e2e/common/createCitizenUser/idamApiHelper.ts

```diff
@@ -0,0 +1,109 @@
+import { APIRequestContext, request } from "@playwright/test";
+import * as dotenv from "dotenv";
+import qs from "qs";
+import { v4 as uuidv4 } from "uuid";
+import { existsSync } from 'fs';
+
+dotenv.config();
+
+/**
+ * Initializes the Playwright API request context.
+ * @returns {Promise<APIRequestContext>}
+ */
+export async function initializeAPIContext(): Promise<APIRequestContext> {
+  return await request.newContext();
+}
+
+/**
+ * Function to get an access token from the IDAM service
+ * @param {APIRequestContext} apiContext The API request context
+ * @returns {Promise<string | null>} The access token if successful, otherwise null
+ */
+export async function getAccessToken(
+  apiContext: APIRequestContext,
+): Promise<string | null> {
+  try {
+    const data = {
+      grant_type: "client_credentials",
+      client_id: "prl-cos-api",
+      client_secret: process.env.IDAM_SECRET,
+      scope: "profile roles",
+    };
+    const response = await apiContext.post(
+      process.env.IDAM_TOKEN_URL as string,
+      {
+        headers: { "content-type": "application/x-www-form-urlencoded" },
+        data: qs.stringify(data),
```

Comment:
You can probably just use JSON.stringify(data) instead of using the extra `qs` library just for this

Reply:
i need to covert the data to a URL-encoded string format (application/x-www-form-urlencoded) to get the access token so not sure i can use json.stringify here (https://tools.hmcts.net/confluence/display/SISM/Creating+Test+Users+in+IDAM)

Reply:
Ah sorry - in that case you can provide form instead of data e.g. https://playwright.dev/python/docs/api/class-apirequestcontext#api-request-context-get-option-form